pkg/cliplugins/workspace/cmd/workspace.go

```diff
@@ -0,0 +1,570 @@
+		Short:        "Uses the given workspace as the current workspace. Using - means previous workspace",
+		Example:      "kcp workspace use my-worspace",
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return fmt.Errorf("The workspace name (or -) should be given")
+			}
+
+			if err := o.Validate(); err != nil {
+				return err
+			}
+
+			if err := o.RunUse(args[0]); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	currentCmd := &cobra.Command{
+		Use:          "current",
+		Short:        "Returns the name of the current workspace",
+		Example:      "kcp workspace current",
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Validate(); err != nil {
+				return err
+			}
+
+			if err := o.RunCurrent(); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	listCmd := &cobra.Command{
+		Use:          "list",
+		Short:        "Returns the list of the personal workspaces of the user",
+		Example:      "kcp workspace list",
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Validate(); err != nil {
+				return err
+			}
+
+			if err := o.RunList(); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	inheritFromFlag := "inheritFrom"
+	useFlag := "use"
+	createCmd := &cobra.Command{
+		Use:          "create",
+		Short:        "Creates a new personal workspace",
+		Example:      "kcp workspace create <workspace name>",
+		SilenceUsage: true,
+		Args:         cobra.ExactArgs(1),
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Validate(); err != nil {
+				return err
+			}
+
+			useAfterCreation, err := c.Flags().GetBool(useFlag)
+			if err != nil {
+				return err
+			}
+			inheritFrom, err := c.Flags().GetString(inheritFromFlag)
+			if err != nil {
+				return err
+			}
+			if err := o.RunCreate(args[0], useAfterCreation, inheritFrom); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	createCmd.Flags().String(inheritFromFlag, "admin", "Specifies another workspace it should inherit CRDs from")
+	createCmd.Flags().Bool(useFlag, false, "Use the new workspace after a successful creation")
+
+	deleteCmd := &cobra.Command{
+		Use:          "delete",
+		Short:        "Deletes a personal workspace",
+		Example:      "kcp workspace delete <workspace name>",
+		SilenceUsage: true,
+		Args:         cobra.ExactArgs(1),
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Validate(); err != nil {
+				return err
+			}
+
+			if err := o.RunDelete(args[0]); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	cmd.AddCommand(useCmd)
+	cmd.AddCommand(currentCmd)
+	cmd.AddCommand(listCmd)
+	cmd.AddCommand(createCmd)
+	cmd.AddCommand(deleteCmd)
+	return cmd, nil
+}
+
+func (o *WorkspaceOptions) Complete(cmd *cobra.Command) error {
+	if err := metav1.AddMetaToScheme(scheme.Scheme); err != nil {
```

Comment:
why this? Modifying somebody else' scheme is not good

Reply:
The `TableXXX` types that are in the meta v1 scheme are not added by default to the generated KCP clientset scheme. So that when I want to get the result of a request done with this clientset as a table, it doesn't know the `Table` types.
It happens that in the `kubectl` code, those types are added to the main `kubectl` scheme.

Reply:
Any suggestion to do better ?